src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/prisonperson/PrisonPersonService.kt

```diff
@@ -101,17 +101,28 @@ class PrisonPersonService(
       throw NotFoundException("No offender found for $offenderNo")
     }
 
-    val latestBooking = bookingRepository.findLatestByOffenderNomsId(offenderNo)
-      ?: throw NotFoundException("No bookings found for $offenderNo")
-
-    return latestBooking.physicalAttributes.findLastModified()
-      ?.let {
-        PrisonPersonReconciliationResponse(
-          offenderNo = offenderNo,
-          height = it.getHeightInCentimetres(),
-          weight = it.getWeightInKilograms(),
-        )
-      }
+    val bookings = bookingRepository.findAllByOffenderNomsId(offenderNo)
+    if (bookings.isEmpty()) {
+      throw NotFoundException("No bookings found for $offenderNo")
+    }
+
+    val latestBookingWithPhysicalAttributes =
+      bookings
+        .filter { it.physicalAttributes.isNotEmpty() }
+        // a null end date is considered the latest
+        .maxByOrNull { it.bookingEndDate ?: LocalDateTime.MAX }
+
+    val physicalAttributes = latestBookingWithPhysicalAttributes
+      ?.physicalAttributes
+      ?.maxBy { it.modifyDatetime ?: it.createDatetime }
+
+    return physicalAttributes?.let {
+      PrisonPersonReconciliationResponse(
+        offenderNo = offenderNo,
+        height = physicalAttributes.getHeightInCentimetres(),
+        weight = physicalAttributes.getWeightInKilograms(),
+      )
```

Comment:
This is how DPS work out the current height and weight - get the last booking with physical attributes, where a null end date is considered the latest.